### PR TITLE
Adding of AWS provider type.

### PR DESCRIPTION
**Pull request by martinpovolny**
Add provider type AWS to the "New Source" form.

### CODE REVIEW

File: src/SmartComponents/ProviderPage/providerForm.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const providerForm = {
 import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
 export const wizardForm = {
     initialValues: {
-        role: 'kubernetes',
+        role: 'kubernetes', // 'aws' for AWS EC2
```
Comment (agrare) [inline src/SmartComponents/ProviderPage/providerForm.js]: Can this be encoded into the source_type schema ?

File: src/SmartComponents/ProviderPage/providerForm.js [hunk 2]
```diff
@@ -37,9 +37,9 @@ export const wizardForm = {
                 name: 'step-1',
                 stepKey: 1,
                 nextStep: {
-                    when: 'source-type',
+                    when: 'source_type',
```
Comment (agrare) [inline src/SmartComponents/ProviderPage/providerForm.js]: What is this case handling that can't be gotten from the source_type records?

**Comment by agrare** [comment]
@martinpovolny the case statements on the hard-coded source_type names are surprising I thought we were trying to get away from that kind of thing?

**Comment by martinpovolny** [comment]
Yes, we where. Therefor I was suggesting to have the creating logic on the server side and on client side.

I will try to first isolate and get rid of any hard-coded provider logic anyway.

File: src/SmartComponents/ProviderPage/providerForm.js (hunk 2 above)
Comment (martinpovolny) [reply src/SmartComponents/ProviderPage/providerForm.js]: This will be gotten from the source_type records in the future.

Comment (agrare) [reply src/SmartComponents/ProviderPage/providerForm.js]: Okay can we do this as a follow up now?  I don't think there was any urgency to get this in immediately but could be wrong

File: src/SmartComponents/ProviderPage/providerForm.js (hunk 1 above)
Comment (martinpovolny) [reply src/SmartComponents/ProviderPage/providerForm.js]: I used the initialValues to pass in the role for the Endpoint.

The problem is that the `initialValues` here are for the whole form. Not a single choice of provider. So in order to not have this hard-coded I need to find a different way to get it from the schema.

Until I find such way I added a case with the provider-specific values including the value for the `role`.

**Pull request merged**